onnxruntime/python/torch_cpp_extensions/aten_op_executor/aten_op_executor.cc

```diff
@@ -78,22 +60,26 @@ struct ATenOperator {
       case c10::TypeKind::IntType: {
         TORCH_INTERNAL_ASSERT(dlpack->dl_tensor.dtype.code == DLDataTypeCode::kDLInt &&
                               dlpack->dl_tensor.dtype.bits == 64);
-        i_value = is_list ? ToListIValue<int64_t>(dlpack, is_optional) : ToIValue<int64_t>(dlpack, is_optional);
+        i_value = is_list ? ToListIValue<int64_t, int64_t>(dlpack, is_optional)
+                          : ToIValue<int64_t, int64_t>(dlpack, is_optional);
       } break;
       case c10::TypeKind::FloatType: {
         TORCH_INTERNAL_ASSERT(dlpack->dl_tensor.dtype.code == DLDataTypeCode::kDLFloat &&
                               dlpack->dl_tensor.dtype.bits == 32);
-        i_value = is_list ? ToListIValue<float>(dlpack, is_optional) : ToIValue<float>(dlpack, is_optional);
+        // PyTorch's IValue doesn't support float, so we convert it to double.
```

Comment:
Does pytorch perform computation with doubles for non-tensor `IValue`s?

Reply:
The computation here for upsample_nearest still take float list, but we need to pass arguments by IValue to them. PyTorch will do the float to double conversion when calling the function.

Reply:
> PyTorch will do the float to double conversion when calling the function.

you mean double to float right?

Reply:
Right...